pipeline/load.py

```diff
@@ -0,0 +1,220 @@
+        return None
+
+
+def get_cursor(conn: connection) -> cursor:
+    """Creates a cursor based on provided psycopg2 connection"""
+    logging.info("Creating a cursor from the psycopg2 connection")
+    try:
+        cursor = conn.cursor(cursor_factory=psycopg2.extras.RealDictCursor)
+        logging.info("Cursor created successfully")
+        return cursor
+    except OperationalError as e:
+        logging.error(
+            f"OperationalError occurred while trying to create a cursor: {e}")
+        return None
+    except Exception as e:
+        logging.error(
+            f"An unexpected error occurred in creating a cursor: {e}")
+        return None
+
+
+def execute_query(cursor: cursor, query: str) -> list:
+    """Executes a query and returns the results."""
+    try:
+        cursor.execute(query)
+        return cursor.fetchall()
+    except OperationalError as e:
+        logging.error(f"OperationalError occurred while executing query: {e}")
+    except Exception as e:
+        logging.error(
+            f"An unexpected error occurred while executing query: {e}")
+    return None
+
+
+def execute_insert(conn: connection, cursor: cursor, query: str, args: tuple, column_wanted: str) -> int:
+    """Executes an insert query and returns the new ID"""
+    try:
+        cursor.execute(query, args)
+        new_id = cursor.fetchone()[column_wanted]
+        conn.commit()
+        return new_id
+    except (psycopg2.IntegrityError, psycopg2.OperationalError, psycopg2.DatabaseError) as e:
+        logging.error(f"Database error: {e}")
+        conn.rollback()
+    except Exception as e:
+        logging.error(
+            f"Unexpected error occurred while adding to the database: {e}")
+        conn.rollback()
+    return None
+
+
+def fetch_all(cursor: cursor, table: str, key_field: str, value_field: str) -> dict:
+    """Gets all records from a given table and returns a dictionary"""
+    logging.info(f"Fetching all values from {table}")
+    query = f"SELECT * FROM {table};"
+    results = execute_query(cursor, query)
+    return {record[key_field]: record[value_field] for record in results} if results else {}
+
+
+def get_all_alerts(cursor: cursor) -> dict:
+    "Fetches all alert values from RDS"
+    return fetch_all(cursor, "alerts", "alert_value", "alert_id")
+
+
+def get_all_networks(cursor: cursor) -> dict:
+    "Fetches all network values from RDS"
+    return fetch_all(cursor, "networks", "network_name", NETWORK_ID)
+
+
+def get_all_statuses(cursor: cursor) -> dict:
+    "Fetches all statuses from RDS"
+    return fetch_all(cursor, "statuses", STATUS, "status_id")
+
+
+def get_all_magtypes(cursor: cursor) -> dict:
+    "Fetches all magtypes values from RDS"
+    return fetch_all(cursor, "magtypes", "magtype_value", MAGTYPE_ID)
+
+
+def get_all_types(cursor: cursor) -> dict:
+    "Fetches all earthquake types from RDS"
+    return fetch_all(cursor, "types", "type_value", TYPE_ID)
```

Comment:
Is there a way to combine these into 1 function and take in alerts/networks etc as a parameter instead?

Reply:
There is, but I don't think it's worth the hassle since we have to have different arguments each time anyways